### PR TITLE
dev-1.2.0

### PR DESCRIPTION
**Pull request by philipobenito**

### CODE REVIEW

File: tests/ContainerTest.php [hunk 1]
```diff
@@ -435,6 +592,54 @@ public function testContainerResolvesRegisteredCallable()
...
+    }
+
+    /**
+     * Asserts that a class method is registered as a callable when aliased as string
+     *
+     * @return void
+     */
+    public function testContainerResolvesRegisteredClassMethodWhenRegisteredAsString()
+    {
+        $c = new Container;
```
Comment (philsturgeon) [inline tests/ContainerTest.php]: As you're requiring PHP 5.4 you could do `(new Container)->` type stuff.

Comment (philipobenito) [inline tests/ContainerTest.php]: @philsturgeon not in this particular case as the method called doesn't return the container but a definition. However, there are probably a couple of places I may be able to do so, next iteration will be splitting the tests out in to multiple files so will look through as part of that.

File: tests/ContainerTest.php [hunk 2]
```diff
@@ -303,7 +406,12 @@ public function testExtendReturnsDefinitionForModificationWhenCalledWithAValidSe
         $this->assertSame($definition, $extend);
     }
 
-    public function testCallExecutesAnonymousFunction()
+    /**
+     * Asserts that call invokes closure
+     *
+     * @return void
```
Comment (philsturgeon) [inline tests/ContainerTest.php]: I dunno if you need to be that specific on tests. void is inferred anyhow.

File: tests/ContainerTest.php (hunk 1 above)
Comment (philsturgeon) [inline tests/ContainerTest.php]: Oh yeah, commenting on a diff doesn't give me the full picture. Sorry!

File: tests/ContainerTest.php [hunk 3]
```diff
@@ -435,6 +592,54 @@ public function testContainerResolvesRegisteredCallable()
...
+     * Asserts that an exception is thrown when attempting to register a non
+     * callable
+     *
+     * @return void
+     */
+    public function testExceptionThrownWhenRegisteringNonCallable()
+    {
+        $this->setExpectedException('InvalidArgumentException');
+
+        $c = new Container;
```
Comment (philsturgeon) [inline tests/ContainerTest.php]: I think I saw this one first, then scrolled around. If I made that `(new Container)->` here I would have been right. :)

Comment (philipobenito) [inline tests/ContainerTest.php]: Yeah, definitely here, will pick all these up.

**Pull request merged**